packages/typespec-test/test/openai_modular/generated/typespec-ts/src/api/operations.ts

```diff
@@ -283,105 +265,63 @@ export async function _getChatCompletionsDeserialize(
-                  ? undefined
-                  : {
-                      severity: p.content_filter_results?.sexual?.["severity"],
-                      filtered: p.content_filter_results?.sexual?.["filtered"],
-                    },
-                violence: !p.content_filter_results?.violence
-                  ? undefined
-                  : {
-                      severity:
-                        p.content_filter_results?.violence?.["severity"],
-                      filtered:
-                        p.content_filter_results?.violence?.["filtered"],
-                    },
-                hate: !p.content_filter_results?.hate
-                  ? undefined
-                  : {
-                      severity: p.content_filter_results?.hate?.["severity"],
-                      filtered: p.content_filter_results?.hate?.["filtered"],
-                    },
-                selfHarm: !p.content_filter_results?.self_harm
-                  ? undefined
-                  : {
-                      severity:
-                        p.content_filter_results?.self_harm?.["severity"],
-                      filtered:
-                        p.content_filter_results?.self_harm?.["filtered"],
-                    },
-              },
-        })),
-    usage: {
-      completionTokens: result.body.usage["completion_tokens"],
-      promptTokens: result.body.usage["prompt_tokens"],
-      totalTokens: result.body.usage["total_tokens"],
-    },
-  };
+  let deserializedResponse: unknown = result.body;
+  deserializedResponse = reshape(
+    deserializedResponse,
+    "created",
+    (value) => new Date(value as string)
+  );
+  deserializedResponse = reshape(
+    deserializedResponse,
+    "choices[].message.function_call",
+    "functionCall"
+  );
+  deserializedResponse = reshape(
+    deserializedResponse,
+    "choices[].finish_reason",
+    "finishReason"
+  );
+  deserializedResponse = reshape(
+    deserializedResponse,
+    "choices[].content_filter_results",
+    "contentFilterResults"
+  );
+  deserializedResponse = reshape(
+    deserializedResponse,
+    "choices[].content_filter_results.self_harm",
+    "selfHarm"
+  );
+  deserializedResponse = reshape(
+    deserializedResponse,
+    "prompt_annotations",
+    "promptFilterResults"
+  );
+  deserializedResponse = reshape(
+    deserializedResponse,
+    "prompt_annotations[].prompt_index",
+    "promptIndex"
+  );
+  deserializedResponse = reshape(
+    deserializedResponse,
+    "prompt_annotations[].content_filter_results",
+    "contentFilterResults"
+  );
+  deserializedResponse = reshape(
+    deserializedResponse,
+    "usage.completion_tokens",
+    "completionTokens"
+  );
+  deserializedResponse = reshape(
+    deserializedResponse,
+    "usage.prompt_tokens",
+    "promptTokens"
+  );
+  deserializedResponse = reshape(
+    deserializedResponse,
+    "usage.total_tokens",
+    "totalTokens"
+  );
+  return deserializedResponse as ChatCompletions;
```

Comment:
I wonder if we should just pass a function to reshape a specific type in a more general way ? feel like if every property has been renamed, we will still need to traverse the object in the operatoins.ts ? https://github.com/Azure/autorest.typescript/issues/1971 maybe just leverage reshape in this better modularization ? 

/cc @dgetu